packages/colony-starter-angular/src/app/app.component.ts

```diff
@@ -18,65 +12,96 @@ export class AppComponent implements OnInit {
-    this.example();
+    this.start();
   }
 
-  public async example() {
-
+  public async start() {
     // Create an instance of ColonyNetworkClient using the adapter
-    this.state.networkClient[0] = await connectNetwork(0);
+    this.appService.connectNetwork().then( (res) => {
+      console.log(res);
+      this.state.networkClient[0] = res;
+      this.model.networkClient.addr = this.state.networkClient[0]._contract.address;
+      this.model.networkClient.accountAddr = this.state.networkClient[0]._contract.signer.address;
+    });
+  }
 
     // Create an ERC20 token
-    this.state.tokenAddress = await createToken(
-      this.state.networkClient[0],      // networkClient
-      "Token",                          // name
-      "TKN",                            // symbol
-    );
-
-    // Create a colony using the token address of the ERC20 token
-    this.state.colony = await createColony(
-      this.state.networkClient[0],      // networkClient
-      this.state.tokenAddress,          // tokenAddress
-    );
-
-    // Get an initialized ColonyClient for the colony
-    this.state.colonyClient[0] = await getColonyClient(
-      this.state.networkClient[0],      // networkClient
-      this.state.colony.id,             // colonyId
-    );
+    public async token(name, symbol) {
+    this.appService.createToken(this.state.networkClient[0], name, symbol).then( (res) => {
+      console.log(res);
+      this.model.token.name = name;
+      this.model.token.symbol = symbol;
+      this.model.token.address = res;
+    });
+  }
 
-    // Add a domain to the colony
-    this.state.domain = await addDomain(
-      this.state.colonyClient[0],       // colonyClient
-      1,                                // parentDomainId
-    );
+  // Create a colony using the token address of the ERC20 token
+  public async createColony() {
+    this.appService.createColony(this.state.networkClient[0], this.model.token.address).then( (res) => {
+      console.log(res);
+      this.model.colony = res;
+      this.getColonyClient();
+    });
+  }
 
-    // Create a task in the colony
-    this.state.task = await createTask(
-      this.state.colonyClient[0],       // colonyClient
-      this.state.domain.id,             // domainId
-      {
-        description: "Title Description",      // description
-        title: "Task Title",                  // title
-      },
-    );
+  // Get an initialized ColonyClient for the colony
+  public async getColonyClient() {
+    this.appService.getColonyClient(this.state.networkClient[0], this.model.colony.id).then( (res) => {
+      console.log(res);
+      this.state.colonyClient[0] = res;
+    });
+  }
 
+  // Add a domain to the colony
+  public async addDomain() {
+    this.appService.addDomain(this.state.colonyClient[0], this.model.parentDomainId).then( (res) => {
+      console.log(res);
+      this.model.domain = res;
+    });
   }
 
+  // Create a task in the colony
+  public async createTask(title, desc) {
```

Comment:
The `createTask` method should accept `domainId` as a parameter. The user should be able to choose the domain for the task they are creating.

Reply:
See [commit](https://github.com/JoinColony/colonyStarter/pull/60/commits/4830b5bc16549eacc27e7de37ba0490b636198c2)